src/main/java/uk/gov/justice/digital/hmpps/keyworker/rolemigration/RoleAssignmentsService.java

```diff
@@ -32,48 +38,96 @@ public RoleAssignmentsService(RoleService roleService) {
      * Remove from the user (at the caseload) the roles identified by specification.rolesToRemove.
      */
     @PreAuthorize("hasAnyRole('MAINTAIN_ACCESS_ROLES_ADMIN')")
-    public void updateRoleAssignments(@Valid final RoleAssignmentsSpecification specification) {
+    public Map<String, RoleAssignmentStats> updateRoleAssignments(@Valid final RoleAssignmentsSpecification specification) {
         log.info("Updating role assignments: {}", specification);
+        val caseloadResults = new TreeMap<String, RoleAssignmentStats>();
```

Comment:
why a `TreeMap` here - do you mean a `LinkedHashMap`?

Reply:
no longer applicable as using list